delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/DataEncoder.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.indexing
+
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.BNode
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api.{JsonLdApi, JsonLdJavaApi}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue.ContextObject
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.sdk.views.model.IndexingData
+import ch.epfl.bluebrain.nexus.delta.sdk.implicits._
+import io.circe.Json
+import io.circe.syntax.EncoderOps
+import monix.bio.IO
+
+sealed trait DataEncoder
+
+object DataEncoder extends DataEncoder {
+
+  private val defaultContext =
+    ContextValue(contexts.elasticsearchIndexing, contexts.indexingMetadata)
+
+  implicit private[indexing] val api: JsonLdApi = JsonLdJavaApi.lenient
+
+  def defaultEncoder(
```

Comment:
To create the ES document

We still merge the source to the compacted graph and metadata